swift-ci/sdks/android/scripts/fetch-source.sh

```diff
@@ -161,3 +161,26 @@ pushd boringssl >/dev/null 2>&1
 git checkout ${BORINGSSL_VERSION}
 popd >/dev/null 2>&1
 groupend
+
+groupstart "Patchin Sources"
+# This `git grep` invocation in a trunk test fails in our Docker for some
+# reason, so just turn it into a plain `grep` again.
+perl -pi -e 's:"git",:#:' swift-project/swift/test/Misc/verify-swift-feature-testing.test-sh
+
+# Work around swiftlang/swift-driver#1822 for now
+perl -pi -g -we "s#(call rm ... \".\{LIBDISPATCH_BUILD_DIR\}\"\n(\s+)fi\n)#\1\2if [[ -d \"\\\${ANDROID_NDK}\" ]]; then call ln -sf \"\\\${SWIFT_BUILD_PATH}/lib/swift\" \"\\\${ANDROID_NDK}/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/lib\"; fi#" swift-project/swift/utils/build-script-impl
+
+# disable backtrace() for Android (needs either API33+ or libandroid-execinfo, or to manually add in backtrace backport)
+perl -pi -e 's;os\(Android\);os\(AndroidDISABLED\);g' swift-project/swift-testing/Sources/Testing/SourceAttribution/Backtrace.swift
```

Comment:
This patch isn't necessary anymore. Swift Testing uses `dlsym()` to look up `backtrace()` before calling it.

Reply:
Yeah but remove it for main! 😭

Reply:
I mean, we have much bigger problems with Testing, as _any_ Swift package with macros sees strange build errors when cross-compiling, swiftlang/swift-package-manager#8094, so I wouldn't worry about many using Testing until that is fixed. 😉 We will probably have to add doc telling people to only use XCTest on Android until then.

As for this change, rather than putting in logic to check both scheme and tag names that use 6.2 and only applying this change then, let's decide after the next trunk tag, when people can try the new `#available(Android` feature on their code finally, if we want to submit that new feature for 6.2. If not, I guarantee you that I will elide this change for 6.3 alone before the branch in a couple weeks. 👍

Reply:
We definitely don't want to tell people to only use XCTest! Given that Swift Testing is part of the toolchain, there should be no reason a developer who can use XCTest can't use Swift Testing.

But okay, I'll stop complaining until after the next tag.

Reply:
> Given that Swift Testing is part of the toolchain, there should be no reason a developer who can use XCTest can't use Swift Testing.

Given that long-standing issue with cross-compiling macros for tests, I don't see any way around warning Android users off Testing for now.